docs/extensions/garden-api-access.md

```diff
@@ -33,7 +40,99 @@ type: Opaque
+
+When installing a [`ControllerInstallation`](controllerregistration.md), gardenlet creates two secrets in the installation's namespace: a generic garden kubeconfig (`generic-garden-kubeconfig-<hash>`) and a garden access secret (`garden-access-extension`).
+Additionally, it injects `volume`, `volumeMounts`, and two environment variables into all objects in the `apps` and `batch` API group:
+
+- The `GARDEN_KUBECONFIG` environment variable points to the path where the generic garden kubeconfig is mounted.
+  This in turn contains the garden cluster's address and current CA bundle as well as the path where the garden access token is mounted.
+- The `SEED_NAME` environment variable is set to the name of the `Seed` where the extension is installed.
+This is useful for restricting watches in the garden cluster to relevant objects.
+
+For example, a `Deployment` deployed via a `ControllerInstallation` will be mutated as follows:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: gardener-extension-provider-local
+  annotations:
+    reference.resources.gardener.cloud/secret-795f7ca6: garden-access-extension
+    reference.resources.gardener.cloud/secret-d5f5a834: generic-garden-kubeconfig-81fb3a88
+spec:
+  template:
+    metadata:
+      annotations:
+        reference.resources.gardener.cloud/secret-795f7ca6: garden-access-extension
+        reference.resources.gardener.cloud/secret-d5f5a834: generic-garden-kubeconfig-81fb3a88
+    spec:
+      containers:
+      - name: gardener-extension-provider-local
+        env:
+        - name: GARDEN_KUBECONFIG
+          value: /var/run/secrets/gardener.cloud/garden/generic-kubeconfig/kubeconfig
+        - name: SEED_NAME
+          value: local
+        volumeMounts:
+        - mountPath: /var/run/secrets/gardener.cloud/garden/generic-kubeconfig
+          name: garden-kubeconfig
+          readOnly: true
+      volumes:
+      - name: garden-kubeconfig
+        projected:
+          defaultMode: 420
+          sources:
+          - secret:
+              items:
+              - key: kubeconfig
+                path: kubeconfig
+              name: generic-garden-kubeconfig-81fb3a88
+              optional: false
+          - secret:
+              items:
+              - key: token
+                path: token
+              name: garden-access-extension
+              optional: false
+```
+
+The generic garden kubeconfig will look like this:
+
+```yaml
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    certificate-authority-data: LS0t...
+    server: https://garden.local.gardener.cloud:6443
+  name: garden
+contexts:
+- context:
+    cluster: garden
+    user: extension
+  name: garden
+current-context: garden
+users:
+- name: extension
+  user:
+    tokenFile: /var/run/secrets/gardener.cloud/garden/generic-kubeconfig/token
+```
+
+## Using Self-Managed Garden Access
+
+Extensions can also manage their own garden access secrets if required.
```

Comment:
Maybe it makes sense to provide an example for this use case and perhaps example code? It's quite abstract from just reading it here?

Reply:
Right, this is too abstract to be useful. My goal behind this section was only to document the exceptions of the injection mechanism, i.e., skipping objects that already have the `GARDEN_KUBECONFIG` env var. 
Instead of documenting this in its own section, I merged it with the previous one. Folks who have concrete use cases for this will know what to do :D